packages/circuit-ui/components/ColorInput/ColorInput.tsx

```diff
@@ -0,0 +1,163 @@
+import { clsx } from '../../styles/clsx.js';
+import { FieldLabel, FieldLabelText } from '../Field/index.js';
+import { applyMultipleRefs } from '../../util/refs.js';
+
+import styles from './ColorInput.module.css';
+
+export interface ColorInputProps
+  extends Omit<
+    InputProps,
+    | 'ref'
+    | 'type'
+    | 'defaultValue'
+    | 'value'
+    | 'placeholder'
+    | 'maxLength'
+    | 'pattern'
+    | 'renderPrefix'
+    | 'renderSuffix'
+    | 'as'
+  > {
+  /**
+   * A short string that is shown inside the empty input.
+   */
+  placeholder?: string;
+  /**
+   * The value of the input element.
+   */
+  value?: string;
+  /**
+   * The default value of the input element.
+   */
+  defaultValue?: string;
+  /*
+   * Picker label describes the color picker which serves as an alternative
+   * to the hex color input.
+   */
+  pickerLabel: string;
+}
+
+export const ColorInput = forwardRef<InputElement, ColorInputProps>(
+  (
+    {
+      onChange,
+      className,
+      value,
+      defaultValue,
+      pickerLabel,
+      readOnly,
+      'aria-describedby': descriptionId,
+      id,
+      ...props
+    },
+    ref,
+  ) => {
+    const [currentColor, setCurrentColor] = useState<string | undefined>(
+      defaultValue,
+    );
+    const colorDisplayRef = useRef<HTMLDivElement>(null);
+    const colorPickerRef = useRef<InputElement>(null);
+    const pickerId = useId();
+    const hexSymbolId = useId();
+    const inputFallbackId = useId();
+    const inputId = id || inputFallbackId;
+    const descriptionIds = clsx(hexSymbolId, descriptionId);
+
+    const onPickerColorChange: ChangeEventHandler<InputElement> = (e) => {
+      setCurrentColor(e.target.value);
+      if (onChange) {
+        onChange(e);
+      }
+    };
+
+    const onInputChange: ChangeEventHandler<InputElement> = (e) => {
+      if (colorPickerRef.current) {
+        colorPickerRef.current.value = `#${e.target.value}`;
+      }
+      setCurrentColor(`#${e.target.value}`);
+    };
+
+    useEffect(() => {
+      if (colorDisplayRef.current && currentColor) {
+        colorDisplayRef.current.style.backgroundColor = currentColor;
+      }
+    }, [currentColor]);
+
+    // render suffix only once, otherwise if it gets re-rendered on color change
+    // the native color-picker widget might get mistakenly dismissed by the browser
+    const renderSuffix = useCallback(
+      () => (
+        <div
+          className={clsx(styles.suffix)}
+          ref={colorDisplayRef}
+          style={{ backgroundColor: currentColor }}
+        >
+          <FieldLabel htmlFor={pickerId}>
+            <FieldLabelText label={pickerLabel} hideLabel />
+          </FieldLabel>
+          <input
+            type="color"
+            id={pickerId}
+            className={styles['color-input']}
+            onChange={onPickerColorChange}
+            ref={applyMultipleRefs(colorPickerRef, ref)}
+            aria-controls={inputId}
```

Comment:
I'm not sure whether `aria-controls` is valid in this context 🤔

Reply:
Let's drop this one, I also wasn't 100% certain if it's valid.

Reply:
Removed.